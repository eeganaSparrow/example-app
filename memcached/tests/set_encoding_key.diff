001+ Fatal error: Uncaught Error: Call to a member function setEncodingKey() on null in /Users/takuya/example-app/memcached/tests/set_encoding_key.php:6
002+ Stack trace:
003+ #0 {main}
004+   thrown in /Users/takuya/example-app/memcached/tests/set_encoding_key.php on line 6
001- bool(true)
002- bool(true)
003- string(18) "set using encoding"
004- OK
005- bool(true)
006- bool(false)
007- OK
008- bool(true)
009- string(18) "set using encoding"
010- OK
011- bool(true)
012- bool(false)
013- bool(true)
014- string(18) "set using encoding"
015- OK