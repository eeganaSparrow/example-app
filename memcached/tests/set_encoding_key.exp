bool(true)
bool(true)
string(18) "set using encoding"
OK
bool(true)
bool(false)
OK
bool(true)
string(18) "set using encoding"
OK
bool(true)
bool(false)
bool(true)
string(18) "set using encoding"
OK