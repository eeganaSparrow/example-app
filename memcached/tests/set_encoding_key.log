
---- EXPECTED OUTPUT
bool(true)
bool(true)
string(18) "set using encoding"
OK
bool(true)
bool(false)
OK
bool(true)
string(18) "set using encoding"
OK
bool(true)
bool(false)
bool(true)
string(18) "set using encoding"
OK
---- ACTUAL OUTPUT
Fatal error: Uncaught Error: Call to a member function setEncodingKey() on null in /Users/takuya/example-app/memcached/tests/set_encoding_key.php:6
Stack trace:
#0 {main}
  thrown in /Users/takuya/example-app/memcached/tests/set_encoding_key.php on line 6
---- FAILED
