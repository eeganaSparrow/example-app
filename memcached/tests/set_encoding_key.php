<?php
include dirname (__FILE__) . '/config.inc';
$m = memc_get_instance ();
$key = uniqid ('encoding_test_');

var_dump ($m->setEncodingKey("Hello"));
var_dump ($m->set ($key, 'set using encoding'));
var_dump ($m->get ($key));

echo "OK" . PHP_EOL;

# Change the encryption key. The old value will be inaccessible.
var_dump ($m->setEncodingKey("World"));
var_dump ($m->get ($key));

echo "OK" . PHP_EOL;

# Restore the original key to retrieve old values again.
var_dump ($m->setEncodingKey("Hello"));
var_dump ($m->get ($key));

echo "OK" . PHP_EOL;

# With a new encoding key we can still write new values,
# this works as expected with libmemcached 1.0.18 and higher.
var_dump ($m->setEncodingKey("World"));
var_dump ($m->get ($key));
var_dump ($m->set ($key, 'set using encoding'));
var_dump ($m->get ($key));

echo "OK" . PHP_EOL;

?>
