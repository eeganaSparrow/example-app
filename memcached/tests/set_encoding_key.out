Fatal error: Uncaught Error: Call to a member function setEncodingKey() on null in /Users/takuya/example-app/memcached/tests/set_encoding_key.php:6
Stack trace:
#0 {main}
  thrown in /Users/takuya/example-app/memcached/tests/set_encoding_key.php on line 6